bin/cron/pardot_import

```diff
@@ -0,0 +1,212 @@
+  response = Net::HTTP.post_form(uri, {})
+
+  puts "Response: " + response.to_s
+
+  xml = Nokogiri::XML(response.body, &:noblanks)
+  raise "Pardot create user response did not return parsable XML" if xml.nil?
+
+  error_details = xml.xpath('/rsp/err').text
+  raise InvalidApiKeyException if error_details.include? "Invalid API key or user key"
+
+  status = xml.xpath('/rsp/@stat').text
+  raise "Pardot create user response did not include status" if status.nil?
+
+  raise "Pardot create user response failed with status " + status + " " + xml.to_s if status != "ok"
+end
+
+def append_auth_params_to_url(url)
+  "#{url}&api_key=#{$api_key}&user_key=#{CDO.pardot_user_key}"
+end
+
+def main
+  return unless only_one_running?(__FILE__)
+  prospects = []
+
+  # query the conact rolliups
+  PEGASUS_DB[:contact_rollups].where("roles like '%Teacher%' AND pardot_sync_at IS NULL").each do |contact_rollup|
+    puts contact_rollup[:email]
+
+    # Pardot explicitly considers invalid any email addresses that end with a period in the base
+    # (before the @). We have at least one of these, which appears to be a typo.
+    next if contact_rollup[:email].include? ".@"
+
+    # Map database field names and data to Pardot fields
+    prospect = {}
+    MYSQL_TO_PARDOT_MAP.each do |mysql_key, pardot_info|
+      db_value = contact_rollup[mysql_key]
+      if db_value.presence
+        if pardot_info[:multi]
+          # For multi data fields (multiselect,etc ), we set value names as [fieldname]_0, [fieldname]_1, etc
+          values = db_value.split(",")
+          i = 0
+          values.each do |value|
+            prospect["#{pardot_info[:field]}_#{i}"] = value
+            i += 1
+          end
+        else
+          # for single data fields, just set [fieldname] = value
+          prospect[pardot_info[:field]] = db_value
+        end
+      end
+    end
+    # special case: if contact has opted out, set the two different Pardot flavors of opted out
+    # to true. Also, only ever set this to true, otherwise set no value; never set it back
+    # to false. Pardot is the authority on opt-out data, so never reset any opt-out setting it has stored.
+    if contact_rollup[:opted_out] == true
+      prospect[:opted_out] = true
+      prospect[:is_do_not_email] = true
+    end
+
+    # set a custom field to mark in Pardot that this contact was imported from Code Studio
+    prospect[:Imported_from_Code_Studio] = "true"
+    # add this prosect to the batch
+    prospects << prospect
+
+    # As a sniff test, build the URL that would result from our current prospect list
+    # so we can see how long it is
+    url = build_create_batch_prospects_url(prospects)
+
+    # If the URL is longer than an emperically determined maximum safe length, or if
+    # we have hit our max # of prospects allowed by Pardot in one batch API call, go ahead
+    # and submit the batch
+    if url.length > MAX_URL_LENGTH || prospects.size == MAX_PROSPECT_BATCH_SIZE
```

Comment:
Thx, will rename. If we do exceed the length, it will raise a Ruby exception (inside URI constructor). Exceptions in cron jobs all go to Honeybadger, right?